src/inference/src/cnn_network_ngraph_impl.cpp

```diff
@@ -458,6 +459,7 @@ void CNNNetworkNGraphImpl::reshape(const std::map<std::string, ngraph::PartialSh
                 manager.register_pass<ngraph::pass::ConvertMulticlassNmsToMulticlassNmsIE>(false);
                 manager.register_pass<::ngraph::pass::ConvertMatrixNmsToMatrixNmsIE>(false);
                 manager.register_pass<::ngraph::pass::ConvertNMS9ToNMSIEInternal>();
+                manager.register_pass<::ngraph::pass::ConvertGP9ToGPIEInternal>();
```

Comment:
Sorry I havent deeply followed up this , but should we apply the transform for all target devices? Or we should apply it only for GPU plugin?

Reply:
GenerateProposals has dynamic output shape by its nature, which is not supported by GPU Plugin. So we need this conversion exactly in this place to deal with dynamism, it would not be enough to place this conversion to GPU transformation pipeline only.
Regarding support by other devices plugins - ```ngraph::op::internal::GenerateProposalsIEInternal``` is derived from ```op::v9::GenerateProposals```. So if a plugin does not have special support for ```GenerateProposalsIEInternal```, it handles the operation just like ordinary ```GenerateProposals``` operation. Other "Internal" operations here uses the same approach.

Reply:
Okay thanks for the explanation ;)